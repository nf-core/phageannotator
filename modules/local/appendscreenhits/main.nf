process APPENDSCREENHITS {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' :
        'biocontainers/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' }"

    input:
    tuple val(meta), path(mash_screen),  path(assembly_fasta)
    tuple val(meta2), path(reference_fasta)

    output:
    tuple val(meta), path("*.fasta_w_screen_hits.fna.gz")   , emit: assembly_w_screen_hits
    path "versions.yml"                                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    append_screen_hits.py \\
        --reference_fasta $reference_fasta \\
        --mash_screen_results $mash_screen \\
        --assembly_fasta $assembly_fasta \\
        --prefix $prefix \\
        --output ${prefix}.fasta_w_screen_hits.fna

    gzip ${prefix}.fasta_w_screen_hits.fna

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(echo \$(biopython_version.py 2>&1))
        pandas: \$(echo \$(pandas_version.py 2>&1))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.fasta_w_screen_hits.fna
    gzip ${prefix}.fasta_w_screen_hits.fna

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(echo \$(biopython_version.py 2>&1))
        pandas: \$(echo \$(pandas_version.py 2>&1))
    END_VERSIONS
    """
}
