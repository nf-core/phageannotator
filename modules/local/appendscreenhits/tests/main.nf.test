nextflow_process {

    name "Test process: APPENDSCREENHITS"
    script "../main.nf"
    process "APPENDSCREENHITS"


    test("mash_screen.tsv & fasta.gz & reference_fasta.gz") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.pipelines_testdata_base_path + 'modules/local/append_screen_hits/mash_screen_results.tsv', checkIfExists: true),
                    file(params.pipelines_testdata_base_path + 'modules/local/seqkit/seq/assembly.fasta.gz', checkIfExists: true)
                ]
                input[1] = [
                    [ id: 'reference' ],
                    file(params.pipelines_testdata_base_path + 'modules/local/append_screen_hits/reference.fasta.gz', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("mash_screen.tsv & fasta.gz & reference_fasta.gz - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.pipelines_testdata_base_path + 'modules/local/append_screen_hits/mash_screen_results.tsv', checkIfExists: true),
                    file(params.pipelines_testdata_base_path + 'modules/local/seqkit/seq/assembly.fasta.gz', checkIfExists: true)
                ]
                input[1] = [
                    [ id: 'reference' ],
                    file(params.pipelines_testdata_base_path + 'modules/local/append_screen_hits/reference.fasta.gz', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
