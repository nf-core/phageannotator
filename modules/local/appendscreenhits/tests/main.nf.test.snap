{
    "mash_screen.tsv & fasta.gz & reference_fasta.gz - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta_w_screen_hits.fna.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,65b56210fc3597cf5d869e5a1384a1a3"
                ],
                "assembly_w_screen_hits": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta_w_screen_hits.fna.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,65b56210fc3597cf5d869e5a1384a1a3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-20T09:48:08.671698195"
    },
    "mash_screen.tsv & fasta.gz & reference_fasta.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta_w_screen_hits.fna.gz:md5,8ae1a1f526134b2672776111c9f9f210"
                    ]
                ],
                "1": [
                    "versions.yml:md5,65b56210fc3597cf5d869e5a1384a1a3"
                ],
                "assembly_w_screen_hits": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta_w_screen_hits.fna.gz:md5,8ae1a1f526134b2672776111c9f9f210"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,65b56210fc3597cf5d869e5a1384a1a3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-20T10:32:14.343643855"
    }
}
